File: OsmAnd-turn_screen_on/src/main/java/net/osmand/turn_screen_on/helpers/AndroidUtils.java
package net.osmand.turn_screen_on.helpers;

import android.content.Context;
import android.content.res.Resources;
import android.util.TypedValue;

import static android.util.TypedValue.COMPLEX_UNIT_DIP;

public class AndroidUtils {
    public static int dpToPx(Context ctx, float dp) {
        Resources r = ctx.getResources();
        return (int) TypedValue.applyDimension(
                COMPLEX_UNIT_DIP,
                dp,
                r.getDisplayMetrics()
        );
    }
}

File: OsmAnd-telegram/src/main/java/net/osmand/telegramtest/AndroidUtils.kt
package net.osmand.telegramtest

import android.app.Activity
import android.content.Context
import android.content.res.Configuration
import android.view.View
import android.view.inputmethod.InputMethodManager

object AndroidUtils {

    private fun isHardwareKeyboardAvailable(context: Context): Boolean {
        return context.resources.configuration.keyboard != Configuration.KEYBOARD_NOKEYS
    }

    fun softKeyboardDelayed(view: View) {
        view.post {
            if (!isHardwareKeyboardAvailable(view.context)) {
                val imm = view.context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager?
                imm?.showSoftInput(view, InputMethodManager.SHOW_IMPLICIT)
            }
        }
    }

    fun hideSoftKeyboard(activity: Activity, input: View?) {
        val inputMethodManager = activity.getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager?
        if (inputMethodManager != null) {
            if (input != null) {
                val windowToken = input.windowToken
                if (windowToken != null) {
                    inputMethodManager.hideSoftInputFromWindow(windowToken, 0)
                }
            }
        }
    }
}
